battle-engine.js

```diff
@@ -601,6 +601,22 @@ BattlePokemon = (() => {
 	};
 	BattlePokemon.prototype.transformInto = function (pokemon, user, effect) {
 		let template = pokemon.template;
+	
+		if ((user && user.transformed && this.battle.gen >= 5)&&(BattleSide.prototype.chooseMove(willMega))) {
+			this.transformed = false;
+			this.types = this.types
```

Comment:
not sure i understand the logic here

you also forgot a semi colon on this line